modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/XmlExtensionParserUtils.java

```diff
@@ -0,0 +1,491 @@
+            DataQualifierVisitor visitor = new SimpleTypeDataQualifierVisitor()
+            {
+
+                @Override
+                protected void onSimpleType()
+                {
+                    if (conversionService.canConvert(value.getClass(), expectedDataType.getRawType()))
+                    {
+                        resolverValueHolder.set(new StaticValueResolver(conversionService.convert(value, expectedDataType.getRawType())));
+                    }
+                    else
+                    {
+                        defaultOperation();
+                    }
+                }
+
+                @Override
+                protected void defaultOperation()
+                {
+                    resolverValueHolder.set(new CachingValueResolverWrapper(new RegistryLookupValueResolver(value.toString())));
+                }
+            };
+
+            expectedDataType.getQualifier().accept(visitor);
+            return resolverValueHolder.get();
+        }
+
+        return null;
+    }
+
+    /**
+     * parses a pojo which type is described by {@code pojoType},
+     * recursively moving through the pojo's properties.
+     *
+     * @param element          the XML element which has the bean as a child
+     * @param fieldName        the name of the field in which the parsed pojo is going to be assigned
+     * @param childElementName the name of the the bean's top level XML element
+     * @param pojoType         a {@link DataType} describing the bean's type
+     * @return a {@link org.springframework.beans.factory.config.BeanDefinition} if the bean could be parsed, {@code null}
+     * if the bean is not present on the XML definition
+     */
+    private static ValueResolver parsePojo(ElementDescriptor element,
+                                           String fieldName,
+                                           String childElementName,
+                                           DataType pojoType,
+                                           Object defaultValue)
+    {
+
+        // check if the pojo is referenced as an attribute
+        ValueResolver resolver = getResolverFromAttribute(element, fieldName, pojoType, defaultValue);
+
+        if (resolver != null)
+        {
+            return resolver;
+        }
+
+        // check if the pojo is defined inline as a child element
+        ElementDescriptor childElement = element.getChildByName(childElementName);
+
+        if (childElement != null)
+        {
+            return getPojoValueResolver(pojoType, childElement);
+        }
+
+        // last chance, check if this is a top level element
+        if (getGlobalPojoTypeName(pojoType).equals(element.getName()))
+        {
+            return getPojoValueResolver(pojoType, element);
+        }
+
+        // pojo was not specified, take the coward's route and return a null resolver
+        return new StaticValueResolver(null);
```

Comment:
if the pojo was not specified it might not necessarily be an error, it might just mean that the pojo was optional and not provided.

As for the NullValueResolver, I thought about that many times, but what didn't convinced me is that in cases like this it's pretty straight forward to do NullValueResolver.getInstance(), but in some other cases you would be forced to either:

* use an if or a ternary expression
* have some StaticValueResolver instances return null anyway, which would defeat the purpose